File: Nunit.Test/Examples/System/Collections/ArrayExamples.cs
using NUnit.Framework;

namespace Examples.Examples.System.Collections
{
	[TestFixture]
	public class ArrayExamples
	{
		[Test]
		public void StringArrayDeclaredWithValues() {
			var myArray = new[] { "a", "b" };
			Assert.That(myArray.Length, Is.GreaterThan(1));
			Assert.That(myArray, Is.TypeOf<string[]>());
		}

		[Test]
		public void IntArrayDeclaredWithValues() {
			var myArray = new [] { 1, 2, 3 };
			Assert.That(myArray.Length, Is.GreaterThan(1));
			Assert.That(myArray, Is.TypeOf<int[]>());
		}
	}
}

File: NUnit.vb.Test/Examples/ArrayExamples.vb
Imports NUnit.Framework

Namespace Examples

	<TestFixture>
	Public Class ArrayExamples

		<Test>
		Public Sub CallingIndexOutsideArrayGivesIndexOutOfRangeException()
			Dim theArray = {String.Empty, String.Empty}

			Assert.Throws(Of IndexOutOfRangeException)(Sub()
														   Dim test = theArray(3)
													   End Sub)

		End Sub

	End Class

End Namespace